File: E-commerce-multivendeur-main/src/app/Models/product.ts
import { Category } from "../Models/category";

export interface Product {
  id?: any;                 // facultatif pour la création
  name: string;
  price: number;
  images: string[];             // liste d'URLs des images
  category: Category;           // ou juste categoryId: number;
  description: string;
  nouveautes: boolean;
  remise: boolean;
}

File: meubleBackend/src/main/java/com/example/meublebackend/entity/Product.java
package com.example.meublebackend.entity;
import com.example.meublebackend.converter.StringListConverter;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(columnDefinition = "TEXT") // ou VARCHAR(1000) selon ta BDD
    @Convert(converter = StringListConverter.class)
    private List<String> images;
    private Integer quantite;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    private String description;

    private Boolean nouveautes;

    private Boolean remise;
}
